user-service/src/main/java/rs/edu/raf/banka1/services/CustomerServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+        // Check if the user is authenticated
+        if (authentication != null && authentication.getPrincipal() instanceof UserDetails) {
+            // Assuming your UserDetails implementation has the email field
+            UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+            String email = userDetails.getUsername();
+            employee = userRepository.findByEmail(email).orElse(null);
+            if(employee == null){
+                return null;
+            }
+
+            Customer customer = CustomerMapper.customerDataToCustomer(createCustomerRequest.getCustomerData());
+            String activationToken = UUID.randomUUID().toString();
+            customer.setActivationToken(activationToken);
+            customer = customerRepository.save(customer);
+
+            BankAccount bankAccount = BankAccountMapper.generateBankAccount(
+                    createCustomerRequest.getAccountData(), currency,
+                    customer, employee.getUserId(), createCustomerRequest.getAccountData().getMaintenanceCost());
+
+            bankAccount.setAccountNumber(generateBankAccountNumber());
+            bankAccountRepository.save(bankAccount);
+
+            String to = customer.getEmail();
+            String subject = "Account activation";
+            String text = bankAccount.getAccountNumber();
+            emailService.sendActivationEmail(to, subject, text);
+            return customer.getUserId();
+        }
+        return null;
+    }
+
+    @Override
+    public boolean initialActivation(InitialActivationRequest createCustomerRequest) {
+        BankAccount bankAccount = bankAccountRepository
+                .findBankAccountByAccountNumber(createCustomerRequest.getAccountNumber())
+                .orElse(null);
+        if(bankAccount == null){
+            return false;
+        }
+        if(bankAccount.getCustomer().getEmail().equals(createCustomerRequest.getEmail())
+                && bankAccount.getCustomer().getPhoneNumber().equals(createCustomerRequest.getPhoneNumber())){
+            String to = bankAccount.getCustomer().getEmail();
+            String subject = "Account activation";
+            String text = "localhost:4200/activateCustomer/" + bankAccount.getCustomer().getActivationToken(); //TODO: napravi da radi sa env var
+            emailService.sendActivationEmail(to, subject, text);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Long activateNewCustomer(String token, String password) {
+        Customer customer = customerRepository.findCustomerByActivationToken(token).orElse(null);
+        if(customer == null){
+            return null;
+        }
+        customer.setActivationToken(null);
+        customer.setActive(true);
+        customer.setPassword(passwordEncoder.encode(password));
+        customer = customerRepository.save(customer);
+        BankAccount bankAccount = customer.getAccountIds().get(0);
```

Comment:
da li je bolja opcija gledati za racun da li je aktivan u slucaju da je korisnik aktivan? Ovo mi ima u slucaju da je odgovor na prvi komentar "da".

Reply:
to ima smisla, ali po specifikaciji racun ima polje status, verovatno moze da se desi scenario da je aktivan korisnik a ne racun.